pulsar/producer_partition.go

```diff
@@ -89,7 +88,7 @@ type partitionProducer struct {
 	state  int32
 	client *client
 	topic  string
-	log    *log.Entry
+	logger log.Logger
```

Comment:
Looks like excessive rename. `log` field is OK too.

Reply:
`log` is usually used for the package name, the logger is the one who does the logging job, i think it's more accurate using `logger` than `log`

Reply:
It's true, but it makes review scope unfortunately bigger :(

Reply:
`log` were selected initially by package maintainers, IMO there is no reason to change naming in scope of this PR.
It is really hard to review.

Reply:
Yeah, renaming makes the review scope bigger, but i don't think it's hard to review, it's just a little annoying. The field name using `log` really confuses me, perhaps it's better to open another PR for naming refactor, how about putting a TODO marker first?

Reply:
TODO marker is ok